app/controllers/microservices_engine/v1/data_controller.rb

```diff
@@ -34,18 +34,25 @@ def update
         data = params['content']
         if data.present?
           data.each do |endpoint|
-            existing = Connection.where(object: endpoint['object'])
+            existing = Connection.where(object: endpoint['object']).first
             if existing.present?
               if endpoint['url'].present?
                 # URL exists so we will update as usual
-                existing.update_attributes(name: endpoint['name'], url: endpoint['url'])
+                existing.update_attributes(
+                  name: endpoint.require(:name),
+                  url: endpoint.require(:url)
+                )
               else
                 # URL is blank, thus the endpoint no longer exists
```

Comment:
For simplicity, let's rebuild it every time.

> It'll just get... intensive when/if we get a ton of connection objects like how r3 has a ton of permissions.

This will never happen. This app will have `n` objects where `n` is the number of models which currently live in round-three.

Consider that for each of those models, we have a controller which has on average `c` endpoints, strictly greater than 1. And let's say, given a controller, that on average `g` groups have access to it, strictly greater than 1. Then round three has `ncg` endpoints, strictly greater than `n`.

So since your excellent work with permissions has stopped rebuilding the table every time from being prohibitive over there, I'm sure it'll be fine over here.